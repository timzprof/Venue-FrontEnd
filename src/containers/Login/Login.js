import React, {Component, Fragment} from "react";
import Header from "../../components/Header/Header";
import Footer from "../../components/Footer/Footer";
import Card from "../../components/UI/Card";
import Form from "../../components/Form/Form";
import {updateObject, checkValidity} from "../../helpers/utility";
import classes from "./Login.module.css";

class Login extends Component {
	state = {
		controls: {
			email: {
				formAttributes: {
					type: "email",
					name: "email",
					value: ""
				},
				rules: {
					isEmail: true,
					required: true
				}
			},
			password: {
				formAttributes: {
					type: "password",
					name: "password",
					value: ""
				},
				rules: {
					required: true,
					minLength: 7
				}
			}
		},
		formIsValid: false
	};
	inputChangeHandler = (e, control) => {
		const updatedControls = updateObject(this.state.controls, {
			[control]: updateObject(this.state.controls[control], {
				formAttributes: updateObject(
					this.state.controls[control].formAttributes,
					{
						value: e.target.value
					}
				)
			})
		});
		const formIsValid = checkValidity(
			e.target.value,
			this.state.controls[control].rules
		);

		this.setState({controls: updatedControls, formIsValid});
	};
	submitLogin = () => {};
	render() {
		return (
			<Fragment>
				<Header />
				<section className={classes.Form__container}>
					<Card extraClasses={[classes.Form]}>
						<h3>Login</h3>
						<Form
							controls={this.state.controls}
							inputChangeHandler={this.inputChangeHandler}
							handleSubmit={this.submitLogin}
							submitButton="Login"
						/>
					</Card>
				</section>

				<Footer />
			</Fragment>
		);
	}
}

export default Login;
