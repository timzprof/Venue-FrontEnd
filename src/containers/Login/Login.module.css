.Main {
	min-height: 100vh;
}

.Form {
	background-color: white;
	width: 40%;
	padding: 40px;
	height: fit-content;
}

.Form h3 {
	margin: 10px 5px;
}

.Form button[type="submit"] {
	margin: 5px 0 0 5px;
}

.Form__container {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
	margin: 2rem;
	min-height: calc(100vh - 12rem);
	align-items: center;
}

/* General */
@media only screen and (max-width: 900px) {
	.Form {
		width: 100%;
	}
}

/* Phone */
@media only screen and (max-width: 500px) {
	.Form {
		padding: 30px;
	}
}
