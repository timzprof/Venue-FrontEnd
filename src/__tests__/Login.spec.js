import React from "react";
import Login from "../Pages/login/login";
import {BrowserRouter as Router} from "react-router-dom";
import {create} from "react-test-renderer";

test("Login Container renders", () => {
	const component = create(
		<Router>
			<Login />
		</Router>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
