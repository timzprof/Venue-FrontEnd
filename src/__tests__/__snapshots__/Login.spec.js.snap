// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container renders 1`] = `
Array [
  <header
    className="Header"
  >
    <div
      className="Logo"
    >
      VApp
    </div>
    <a
      className="Link"
      href="/login"
      onClick={[Function]}
    >
      Log In
    </a>
  </header>,
  <section
    className="Form__container"
  >
    <div
      className="Card Form"
    >
      <h3>
        Login
      </h3>
      <form>
        <div
          className="Form_group"
        >
          <input
            className="Form_field"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="Form_group"
        >
          <input
            className="Form_field"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="btn"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </section>,
  <footer>
    <p>
      Â© 2019 Copyright. Venue App.
    </p>
  </footer>,
]
`;
